### PR TITLE
KVM v13.0.0 for Kubernetes 1.18

### PR DESCRIPTION
**Pull request by stone-z**
Towards https://github.com/giantswarm/roadmap/issues/54

### CODE REVIEW

**Comment by tomahawk28** [comment]
Hi, @stone-z 
Could you include [chart-operator 0.13.2](https://github.com/giantswarm/chart-operator/blob/helm2/CHANGELOG.md#v0132-2020-06-23) into the new release? This included several bug fixes for TCs and it would make helm3 migration works smooth in later.

**Comment by tomahawk28** [comment]
@MarcelMue If you are about to release this soon, please consider my earlier comment. https://github.com/giantswarm/releases/pull/450#issuecomment-698955475

File: kvm/v13.0.0/release.yaml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+kind: Release
+metadata:
+  creationTimestamp: null
+  name: v13.0.0
+spec:
+  apps:
+  - name: cert-exporter
+    version: 1.3.0
+  - name: chart-operator
+    version: 0.13.0
```
Comment (cokiengchiara) [inline kvm/v13.0.0/release.yaml:11]:
```suggestion
    version: 0.13.2
```
From @tomahawk28 request here https://github.com/giantswarm/releases/pull/450#issuecomment-698955475 - if it's ok w @MarcelMue !

Comment (MarcelMue) [reply kvm/v13.0.0/release.yaml]: Just FYI we are hitting issues with testing 1.18 - so not sure when this will get merged.

Comment (cokiengchiara) [reply kvm/v13.0.0/release.yaml]: thanks @MarcelMue . Do you think we could do a KVM v12.3.0 (https://github.com/giantswarm/giantswarm/issues/13751) w Helm 3 first?

Comment (MarcelMue) [reply kvm/v13.0.0/release.yaml]: I think you can do that - please coordinate with @giantswarm/team-rocket though. I don't see a big issue though

**Comment by yulianedyalkova** [comment]
I'm not sure why it keeps failing on prow, I tried running cncf manually this morning and they succeeded 🤷‍♀️

**Comment by MarcelMue** [comment]
> I'm not sure why it keeps failing on prow, I tried running cncf manually this morning and they succeeded woman_shrugging

Is it always failing with the upload issue? Is there something I can help debugging with?
I would like to make this more stable :)

**Comment by yulianedyalkova** [comment]
> > I'm not sure why it keeps failing on prow, I tried running cncf manually this morning and they succeeded woman_shrugging
> 
> Is it always failing with the upload issue? Is there something I can help debugging with?
> I would like to make this more stable :)

The last 3 failures are all different.

On the first failure a test failed and then sonobuoy couldn't give info of what's happening:
```
PLUGIN     STATUS   RESULT   COUNT
e2e     failed   failed       1
systemd-logs   complete   passed       4

[2020-11-18T15:43:23Z] Results summary
timeout waiting for results

[2020-11-18T15:43:25Z] Sonobuoy logs
time="2020-11-18T15:43:35Z" level=error msg="could not create sonobuoy client: couldn't get sonobuoy api helper: could not get api group resources: Get \"https://api.fgd7h.k8s.gorgoth.gridscale.kvm.gigantic.io/api?timeout=32s\": net/http: TLS handshake timeout"
```

On the second one the setup got stuck waiting for coredns to come up.

Third one was:
```
Third failure:
[run-tests : run-tests] [2020-11-19T14:08:40Z] Running cncf
[run-tests : run-tests] time="2020-11-19T14:08:40Z" level=error msg="Preflight checks failed"
[run-tests : run-tests] time="2020-11-19T14:08:40Z" level=error msg="namespace already exists"
```

I'll try tro run cncf against the old release to see if it's the installation being borked or the new release.

**Comment by MarcelMue** [comment]
Okay - let me know if I can help somehow.

**Comment by yulianedyalkova** [comment]
It's green 🎉 There is some more context around the previous failures in [this](https://github.com/giantswarm/roadmap/issues/54#issuecomment-731054461) comment and the ones after it.

**Comment by tfussell** [comment]
I have updated a few components. I will retest. Please review the release component versions and release notes.

File: kvm/v13.0.0/release.yaml [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    name: kube-state-metrics
+    version: 1.3.0
+  - componentVersion: 0.3.6
+    name: metrics-server
+    version: 1.1.1
+  - name: net-exporter
+    version: 1.9.2
+  - componentVersion: 1.0.1
+    name: node-exporter
+    version: 1.6.0
```
Comment (MarcelMue) [inline kvm/v13.0.0/release.yaml:27]: This should be bumped to `1.7.0` to prevent issues while upgrading.

Comment (tfussell) [reply kvm/v13.0.0/release.yaml]: Thanks, done.

File: .github/workflows/zz_generated.gitleaks.yaml [hunk 3]
```diff
@@ -0,0 +1,17 @@
+# DO NOT EDIT. Generated with:
```
Comment (tfussell) [inline .github/workflows/zz_generated.gitleaks.yaml:1]: @stone-z Should we remove this file?

File: kvm/v13.0.0/README.md [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+
+
+### kubernetes [1.18.12](https://github.com/kubernetes/kubernetes/releases/tag/v1.18.12)
+
+Kubernetes 1.18 includes a large number of features, deprecations, and bug fixes. We have attempted to summarize the changes relevant to Giant Swarm customers in the following section. The full list of changes can be viewed [here](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.18.md).
+
+Please discuss with your Giant Swarm solution engineer if you are unsure about a cluster's Kubernetes 1.18 upgrade readiness.
+
+#### kube-apiserver
+- The following features are unconditionally enabled and the corresponding `--feature-gates` flags have been removed: `PodPriority`, `TaintNodesByCondition`, `ResourceQuotaScopeSelectors` and `ScheduleDaemonSetPods`.
```
Comment (tfussell) [inline kvm/v13.0.0/README.md:15]: Is this too much detail? Tried to only mention things that could affect users.

File: .github/workflows/zz_generated.gitleaks.yaml (hunk 3 above)
Comment (stone-z) [reply .github/workflows/zz_generated.gitleaks.yaml:1]: I see no harm in having it in `releases` -- WDYT?

File: kvm/v13.0.0/README.md (hunk 4 above)
Comment (stone-z) [reply kvm/v13.0.0/README.md:15]: IMO this is good to know and not too much to include here

File: .github/workflows/zz_generated.gitleaks.yaml (hunk 3 above)
Comment (tfussell) [reply .github/workflows/zz_generated.gitleaks.yaml:1]: Fine with me. We should eventually get it from our `github` dev automation though I guess.

**Comment by tfussell** [comment]
I retested this with all of the new component versions. Creation and upgrades look good. I think we're ready to go once the release notes are checked.

**Review by MarcelMue — APPROVED** [review]
LGTM :rocket:

**Review by cornelius-keller — APPROVED** [review]

**Comment by cornelius-keller** [comment]
LGTM form my side as well.

File: kvm/v13.0.0/release.yaml [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+kind: Release
+metadata:
+  creationTimestamp: null
+  name: v13.0.0
+spec:
+  apps:
+  - name: cert-exporter
+    version: 1.3.0
+  - name: chart-operator
+    version: 2.5.0
```
Comment (cokiengchiara) [inline kvm/v13.0.0/release.yaml:13]: @giantswarm/team-batman-engineers pls confirm this has the bug that we are fixing, but we don't want to release

(sorry Team Rocket 🙈 )

File: kvm/v13.0.0/release.yaml [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+    releaseOperatorDeploy: true
+  - name: calico
+    version: 3.15.3
+  - name: cert-operator
+    releaseOperatorDeploy: true
+    version: 0.1.0
+    reference: 0.1.0-2
+  - name: cluster-operator
+    releaseOperatorDeploy: true
+    version: 0.23.18
```
Comment (cokiengchiara) [inline kvm/v13.0.0/release.yaml:40]: @giantswarm/team-batman-engineers pls confirm this has the bug that we are fixing, but we don't want to release

(sorry Team Rocket 🙈 )

File: kvm/v13.0.0/release.yaml (hunk 5 above)
Comment (rossf7) [inline kvm/v13.0.0/release.yaml:13]:
```suggestion
    version: 2.5.1
```

Yes I'm afraid there is a bug in the status resource and we need to fix the resources.

Patch is already out. 
https://github.com/giantswarm/chart-operator/blob/master/CHANGELOG.md#251---2020-12-01

File: kvm/v13.0.0/release.yaml (hunk 6 above)
Comment (rossf7) [reply kvm/v13.0.0/release.yaml:40]: Yes we need to fix cluster-operator. It's fixed in master and @tomahawk28 is backporting to legacy. We'll post here when the release is out.

Comment (rossf7) [inline kvm/v13.0.0/release.yaml:40]:
```suggestion
    version: 0.23.19
```

cluster-operator release is also out now.

**Review by rossf7 — COMMENTED** [review]
@MarcelMue Patch releases cluster-operator and chart-operator are done and are already used in Azure v13.0.1.

**Review by tfussell — APPROVED** [review]

**Pull request merged**